src/optimizer_interface.jl

```diff
@@ -63,38 +56,19 @@ model = Model()
 set_optimizer(model, GLPK.Optimizer)
 ```
 """
-function set_optimizer(model::Model, optimizer_factory;
+function set_optimizer(model::Model, optimizer_constructor;
                        bridge_constraints::Bool=true)
     error_if_direct_mode(model, :set_optimizer)
-    if !applicable(optimizer_factory)
-        error(_set_optimizer_not_callable_message)
-    end
-    optimizer = optimizer_factory()
-    if !isa(optimizer, MOI.AbstractOptimizer)
-        error("The provided optimizer_factory returned an object of type " *
-              "$(typeof(optimizer)). Expected a " *
-              "MathOptInterface.AbstractOptimizer.")
-    end
-    if !MOI.is_empty(optimizer)
-        error("The provided optimizer_factory returned a non-empty optimizer.")
-    end
     if bridge_constraints
         # The names are handled by the first caching optimizer.
         # If default_copy_to without names is supported, no need for a second
```

Comment:
Is this comment still relevant? The corresponding `if` statement was removed.

Reply:
This is to explain why we have `with_names = false`

Reply:
It doesn't explicitly mention `with_names`. What about formulating it as "We set with_names=false because ..."